src/v/kafka/server/handlers/fetch.cc

```diff
@@ -367,7 +360,7 @@ static ss::future<std::vector<read_result>> fetch_ntps_in_parallel(
       std::move(ntp_fetch_configs),
       [&cluster_pm, &replica_selector, deadline, foreign_read](
         const ntp_fetch_config& ntp_cfg) {
-          auto p_id = ntp_cfg.ntp().tp.partition;
+          auto p_id = ntp_cfg.ntp().get_partition();
```

Comment:
This change will appear repeatedly: we use `get_partition` instead of direct access to the `.partition` member.

The underlying reason is that `ktp_with_hash` cannot safely expose topic or partition as non-`const` members since they couldn't trap modifications which is necessary to update the hash_code. So we have an accessor method instead, which also means we are free to change the internal representation.